### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by therealchriswoodward** [microsoft/microsoft-ui-xaml#8658] [Q&A]

**Comment by DarranRowe** [comment]
What documentation gave this code? I ask because it is outright incorrect if you are trying to use it with WinUI 3. There are three major things in this which are wrong.
1) [CoreApplication](https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.core.coreapplication?view=winrt-22621) is documented to only have the Properties property available in desktop applications.
"This API is supported in Universal Windows Platform apps only, except for the [Properties](https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.core.coreapplication.properties?view=winrt-22621#windows-applicationmodel-core-coreapplication-properties) property, which is always available."
By extension, this means that CoreApplication.MainView will always fail.
2) CoreDispatcher is [documented](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-supported-api#apis-that-have-dependencies-on-uwp-only-ui-features) to be outright unavailable in desktop applications. What's more, CoreDispatcher isn't a dispatcher queue. For WinUI 3, you would be using the Windows App SDK [DispatcherQueue](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.dispatching.dispatcherqueue?view=windows-app-sdk-1.3). This can be obtained from [Microsoft.UI.Xaml.Window.DispatcherQueue](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.window.dispatcherqueue?view=windows-app-sdk-1.3#microsoft-ui-xaml-window-dispatcherqueue).
3) The pickers must use be properly initialised in the desktop environment. This again is [documented](https://learn.microsoft.com/en-us/windows/apps/develop/ui-input/display-ui-objects). In the desktop environment, you must set a desktop window as the parent of the picker, which is what that IInitializeWithWindow does. Be aware that you must obtain the window handle from a class that derives from Microsoft.UI.Xaml.Window.
And just to clarify, WinUI 3 applications only run in the desktop environment. The library and packaging may make them look like they are UWP, but they are not.
